tests/core/order/test.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+    order=(No_order None order0 order_third order_fourth)
+    order_value=(0 0 0 3 4)
+
+    function check_correct_order() {
+        local output=$1
+        for ((i = 0; i < 5; i++)); do
+          rlRun "sed ''"$((i+1))"','"$((i+1))"'!d' $output \
+            | grep \"${order[i]}\""
+        done
+    }
+
+    rlPhaseStartSetup
+        rlRun "output=\$(mktemp)" 0 "Create output file"
+        rlRun "set -o pipefail"
+        rlRun "pushd data"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Order with tmt ls"
+        for tmt in 'tests' 'plans' 'stories'; do
+          rlRun "tmt $tmt ls | tee $output"
+          check_correct_order "$output"
+        done
+    rlPhaseEnd
+
+    rlPhaseStartTest "Order with tmt show"
+        for tmt in 'tests' 'plans' 'stories'; do
+          for ((i = 0; i < 5; i++)); do
+            rlRun "tmt $tmt show \"${order[i]}\" | grep order | tee $output"
+            rlAssertGrep "${order_value[i]}" $output
+          done
+        done
+    rlPhaseEnd
+
+    rlPhaseStartTest "Tests order with tmt run discover plan"
+        rlRun "tmt run -v discover plan -n ${order[3]} -q | grep test | tee $output"
```

Comment:
```suggestion
        rlRun "tmt run -q discover -v plan -n ${order[3]} | grep test | tee $output"
```

I'm not sure what `plan --quite` should do, I'm afraid this is not defined and might change :) Let's rather use what's for sure :)